components/brave_federated/learning_service.cc

```diff
@@ -0,0 +1,202 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/learning_service.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/logging.h"
+#include "base/task/thread_pool.h"
+#include "base/time/time.h"
+#include "base/timer/timer.h"
+#include "base/files/file_util.h"
+#include "brave/components/brave_federated/communication_adapter.h"
+#include "brave/components/brave_federated/eligibility_service.h"
+#include "brave/components/brave_federated/features.h"
+#include "brave/components/brave_federated/task/federated_task_runner.h"
+#include "brave/components/brave_federated/config_utils.h"
+#include "brave/components/brave_federated/task/model.h"
+#include "brave/components/brave_federated/task/typing.h"
+#include "brave/components/brave_federated/util/synthetic_dataset.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+namespace brave_federated {
+
+namespace {
+
+absl::optional<TaskResult> LoadDatasetAndRunTask(
+    std::unique_ptr<FederatedTaskRunner> task_runner) {
+  auto synthetic_dataset = std::make_unique<SyntheticDataset>(500);
+  SyntheticDataset test_dataset = synthetic_dataset->SeparateTestData(50);
+
+  task_runner->SetTrainingData(synthetic_dataset->GetDataPoints());
+  task_runner->SetTestData(test_dataset.GetDataPoints());
+  VLOG(2) << "Model and data set. Task runner initialized.";
+
+  return task_runner->Run();
+}
+
+}  // namespace
+
+LearningService::LearningService(
+    EligibilityService* eligibility_service,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : url_loader_factory_(url_loader_factory),
+      eligibility_service_(eligibility_service) {
+  DCHECK(!init_task_timer_);
+
+  lsc_ = std::unique_ptr<LearningServiceConfig>(new LearningServiceConfig(
+    base::FilePath(FILE_PATH_LITERAL("components/brave_federated/config.json"))));
```

Comment:
How does it work on a end user installation (without the sources)?

Reply:
To make it work, we should bundle this file to the browser resource bundle (see .grd files)